### PR TITLE
feat: min deposit rates query

### PR DESCRIPTION
**Pull request by yruej301**

### CODE REVIEW

File: apps/e2e/src/engine-client/fullSanity.ts [hunk 1]
```diff
@@ -399,6 +402,41 @@ async function fullSanity(context: RunContext) {
...
+  const minDepositRates = await client.getMinDepositRates();
+
+  prettyPrint('Min deposit rates', minDepositRates);
+
+  const subInfo = await client.getSubaccountSummary({
+    subaccountOwner: await signer.getAddress(),
+    subaccountName: 'default',
+  });
+
+  for (const balance of subInfo.balances) {
```
Comment (frankfka) [inline apps/e2e/src/engine-client/fullSanity.ts:415]: nit: We can use `forEach` (a bit more "TS"-y)

File: apps/e2e/src/engine-client/fullSanity.ts [hunk 2]
```diff
@@ -399,6 +402,41 @@ async function fullSanity(context: RunContext) {
     subaccountName: 'default',
   });
   prettyPrint('Subaccount info after withdraw collateral', subaccountInfoAtEnd);
+
+  const minDepositRates = await client.getMinDepositRates();
+
+  prettyPrint('Min deposit rates', minDepositRates);
+
+  const subInfo = await client.getSubaccountSummary({
```
Comment (frankfka) [inline apps/e2e/src/engine-client/fullSanity.ts:410]: To reduce number of calls here, could we just use `subaccountInfoAtEnd`?

File: packages/engine-client/src/utils/queryDataMappers.ts [hunk 3]
```diff
@@ -255,3 +259,25 @@ export function mapEngineMarketPrice(
...
+    baseResponse.min_deposit_rates,
+    mapEngineMinDepositRate,
+  );
+
+  return {
+    minDepositRates,
+  };
+}
+
+export function mapEngineMinDepositRate(
```
Comment (frankfka) [inline packages/engine-client/src/utils/queryDataMappers.ts:276]: nit: we might be able to just do this inline in `EngineQueryClient`, these fns aren't used / relevant anywhere else

File: packages/engine-client/src/utils/queryDataMappers.ts [hunk 4]
```diff
@@ -255,3 +259,25 @@ export function mapEngineMarketPrice(
...
+export function mapEngineServerDepositRates(
+  baseResponse: EngineServerMinDepositRatesResponse,
+): GetEngineMinDepositRatesResponse {
+  const minDepositRates: Record<number, EngineMinDepositRate> = mapValues(
+    baseResponse.min_deposit_rates,
+    mapEngineMinDepositRate,
+  );
+
+  return {
+    minDepositRates,
```
Comment (frankfka) [inline packages/engine-client/src/utils/queryDataMappers.ts:272]: Backend nests in an object, but do we need to do this? or could we just return the `Record<number, EngineMinDepositRate>`?

File: apps/e2e/src/engine-client/fullSanity.ts [hunk 5]
```diff
@@ -399,6 +402,41 @@ async function fullSanity(context: RunContext) {
...
+    if (balance.type == ProductEngineType.SPOT) {
+      const minDepositRate =
+        minDepositRates.minDepositRates[balance.productId].minDepositRate;
+      console.log('product', balance.productId);
+      console.log('min Deposit Rate', minDepositRate.toNumber());
+      console.log(
+        'deposit APR',
+        calcRealizedDepositRateForTimeRange(
+          balance,
+          31536000,
```
Comment (frankfka) [inline apps/e2e/src/engine-client/fullSanity.ts:425]: Could we use `TimeInSeconds`

Comment (yruej301) [reply apps/e2e/src/engine-client/fullSanity.ts:425]: sure

File: packages/engine-client/src/utils/queryDataMappers.ts (hunk 3 above)
Comment (yruej301) [reply packages/engine-client/src/utils/queryDataMappers.ts:276]: gotcha, sg will clean it up

File: apps/e2e/src/engine-client/fullSanity.ts [hunk 6]
```diff
@@ -399,6 +407,36 @@ async function fullSanity(context: RunContext) {
...
+
+  const minDepositRates = await client.getMinDepositRates();
+
+  prettyPrint('Min deposit rates', minDepositRates);
+
+  subaccountInfoAtEnd.balances.forEach((balance) => {
+    if (balance.type == ProductEngineType.SPOT) {
+      const minDepositRate =
+        minDepositRates.minDepositRates[balance.productId].minDepositRate;
+      console.log('product', balance.productId);
```
Comment (frankfka) [inline apps/e2e/src/engine-client/fullSanity.ts:419]: Should we just combine all of these calls? `console.log` takes varargs. You could also just print out an object (can use `toPrintableObject`) - could leave rates as fractions to make life easier too

File: packages/contracts/src/utils/interest.ts [hunk 7]
```diff
@@ -121,12 +122,14 @@ export function calcRealizedDepositRateForTimeRange(
...
 ) {
   const utilization = calcUtilizationRatio(product);
   if (utilization.eq(0)) {
     return toBigDecimal(0);
   }
   return utilization
-    .times(calcBorrowRateForTimeRange(product, seconds))
-    .times(BigDecimals.ONE.minus(toBigDecimal(interestFeeFrac)));
+    .times(calcBorrowRateForTimeRange(product, seconds, toBigDecimal(0)))
+    .times(BigDecimals.ONE.minus(toBigDecimal(interestFeeFrac)))
```
Comment (frankfka) [inline packages/contracts/src/utils/interest.ts:133]: Actually think you might be able to skip these `toBigDecimal` calls

**Review by frankfka — APPROVED** [review]
Lets hold off on merging until the FE work is complete

File: packages/engine-client/src/utils/queryDataMappers.ts (hunk 4 above)
Comment (frankfka) [reply packages/engine-client/src/utils/queryDataMappers.ts]: We are a bit inconsistent with this unfortunately :( ex. `getAllMarkets` just returns `markets` directly

Comment (frankfka) [reply packages/engine-client/src/utils/queryDataMappers.ts]: Looking at this more let's just return `minDepositRates` directly, we can switch the outliers over some other time

**Pull request merged**